server/src/main/java/org/opensearch/action/search/StreamSearchTransportService.java

```diff
@@ -54,6 +55,13 @@ public StreamSearchTransportService(
         this.transportService = transportService;
     }
 
+    public static final Setting<Boolean> STREAM_SEARCH_ENABLED = Setting.boolSetting(
```

Comment:
@rishabhmaurya putting this setting here to not mix with normal search settings (SearchService has a lot). Let me know if you think of a better place.